File: CompPython/b2.py
N, S = list(map(int, input().split()))
mp = {}
direction = 1
size = 1
cnt = 0
visited = [0 for i in range(N)]
for i in range(N):
    q, v = list(map(int, input().split()))
    mp[i] = (q, v)

init = S - 1
while init >= 0 and init < N:
    if mp[init][0] and mp[init][1] <= size and not visited[init]:
        cnt += 1
        visited[init] = 1
    elif mp[init][0] == 0:
        direction *= -1
        size += mp[init][1]
    init += direction * size

print(cnt)

File: CompPython/b2.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

int main()
{
    int N, S;
    cin >> N >> S;

    unordered_map<int, pair<int, int>> mp;
    int direction = 1;
    int size = 1;
    int cnt = 0;
    vector<int> visited(N, 0);

    for (int i = 0; i < N; ++i)
    {
        int q, v;
        cin >> q >> v;
        mp[i] = {q, v};
    }

    int init = S - 1;
    while (init >= 0 && init < N)
    {
        if (mp[init].first && mp[init].second <= size && !visited[init])
        {
            cnt++;
            visited[init] = 1;
        }
        else if (mp[init].first == 0)
        {
            direction *= -1;
            size += mp[init].second;
        }
        init += direction * size;
    }

    cout << cnt << endl;
    return 0;
}